File: Counting_Sort/Counting_Sort.py
def counting_sort(arr):
    """
    Sorts a list of integers using the Counting Sort algorithm.

    Parameters:
    - arr (list): The list to be sorted.

    Returns:
    - list: A sorted list.
    """
    max_value = max(arr)
    count_arr = [0] * (max_value + 1)

    # Count the occurrences of each element in the input list
    for num in arr:
        count_arr[num] += 1

    sorted_arr = []

    # Build the output list using the counting array
    for i in range(len(count_arr)):
        sorted_arr.extend([i] * count_arr[i])

    return sorted_arr


original_list = [64, 25, 12, 22, 11]
sorted_list = counting_sort(original_list)

print("Original List:", original_list)
print("Sorted List:", sorted_list)

File: Counting_Sort/Counting_Sort.ts
/**
 * Sorts an array of integers using the Counting Sort algorithm.
 * @param arr - The array to be sorted.
 * @returns A sorted array.
 */
function countingSort(arr: number[]): number[] {
    const max = Math.max(...arr);
    const countArr: number[] = Array(max + 1).fill(0);

    // Count the occurrences of each element in the input array
    arr.forEach((num) => countArr[num]++);

    const sortedArr: number[] = [];

    // Build the output array using the counting array
    countArr.forEach((count, num) => {
        for (let i = 0; i < count; i++) {
            sortedArr.push(num);
        }
    });

    return sortedArr;
}

// Example usage of Counting Sort
const originalList: number[] = [64, 25, 12, 22, 11];
const sortedList: number[] = countingSort(originalList);

console.log("Original List:", originalList);
console.log("Sorted List:", sortedList);
